background.js

```diff
@@ -1,24 +1,27 @@
 function onBeforeRequest(data){
     //perhaps put 'data' in parameter
     console.log("top of on before request method");
-    console.log(data);
+    console.log(data.initiator);
     //need to return something
     //this is where blocking goes and also figuring out whether something needs to be blocked
-    var urlNotFixed = window.location.toString();
+
+    //rework all of this!
+    var urlNotFixed = data.initiator;
     urlNotFixed.replace(" ", "");
```

Comment:
You shouldn't need to do this once you're working with the right property on the `data` object - you can just lowerecase everything and then run that `includes` test on the whole string.

Reply:
You can use `const` here too. Although you do need to modify the original property on the `data` object by lowercasing it, you can chain the `toLowerCase()` method to the property accessor using the dot syntax you're already using to access the method in the first place, and assign the value of that whole chained expression to a `const`.